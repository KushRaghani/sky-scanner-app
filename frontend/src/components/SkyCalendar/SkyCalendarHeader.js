import React from "react";
import PropTypes from "prop-types";
import { formatDate } from "../../utils/dateUtil";

const SkyCalendarHeader = ({ currentMonth, prevMonth, nextMonth }) => {
  const dateFormat = "Month Year";
  // Rendering the header for the calendar
  return (
    <div className="header calendar-row flex-middle">
      <div className="block block-start">
        <div className="icon" onClick={prevMonth}>
          chevron_left
        </div>
      </div>
      <div className="block block-center">
        <span>{formatDate(currentMonth, dateFormat)}</span>
      </div>
      <div className="block block-end" onClick={nextMonth}>
        <div className="icon">chevron_right</div>
      </div>
    </div>
  );
};

SkyCalendarHeader.propTypes = {
  currentMonth: PropTypes.object.isRequired,
  prevMonth: PropTypes.func.isRequired,
  nextMonth: PropTypes.func.isRequired
};

export default SkyCalendarHeader;
