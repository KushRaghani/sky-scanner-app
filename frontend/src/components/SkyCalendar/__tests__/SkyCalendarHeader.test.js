import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import SkyCalendarHeader from "../SkyCalendarHeader";

function loadSkyCalendarHeader(args) {
  const defaultProps = {
    currentMonth: new Date(),
    prevMonth: () => {},
    nextMonth: () => {}
  };

  const props = { ...defaultProps, ...args };
  return <SkyCalendarHeader {...props} />;
}
describe("SkyCalendarHeader", () => {
  // Enyzme tests
  test("renders SkyCalendarHeader", () => {
    const wrapper = shallow(loadSkyCalendarHeader());
    expect(wrapper.find(".block.block-start").length).toBe(1);
    expect(wrapper.find(".block.block-center").length).toBe(1);
    expect(wrapper.find(".block.block-end").length).toBe(1);
    expect(wrapper.find(".block.block-start").text()).toEqual("chevron_left");
    const date = new Date();
    const headerText =
      date.toLocaleString("default", { month: "long" }) +
      " " +
      date.getFullYear();
    expect(wrapper.find(".block.block-center").text()).toEqual(headerText);
    expect(wrapper.find(".block.block-end").text()).toEqual("chevron_right");
  });

  // Snapshot testing
  test("snapshot with current month", () => {
    const tree = renderer.create(loadSkyCalendarHeader());
    expect(tree).toMatchSnapshot();
  });
});
