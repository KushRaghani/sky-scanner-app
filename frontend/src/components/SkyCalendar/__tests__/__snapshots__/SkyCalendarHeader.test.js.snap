// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkyCalendarHeader snapshot with current month 1`] = `
<div
  className="header calendar-row flex-middle"
>
  <div
    className="block block-start"
  >
    <div
      className="icon"
      onClick={[Function]}
    >
      chevron_left
    </div>
  </div>
  <div
    className="block block-center"
  >
    <span>
      September 2019
    </span>
  </div>
  <div
    className="block block-end"
    onClick={[Function]}
  >
    <div
      className="icon"
    >
      chevron_right
    </div>
  </div>
</div>
`;
